New error extension `http` for setting status/headers in context, resolvers (#6857)
apollographql/apollo-server: docs/source/data/resolvers.mdx

}
```

> If your context function throws an error, Apollo Server will return that error to the user in a JSON response (after invoking your [`formatError` hook](./errors#for-client-responses)) and will not execute the GraphQL operation. By default, this will use an HTTP status code of 500. Additionally, if the thrown error is not a `GraphQLError`, the error's message will be prepended with `"Context creation failed: "`. You can control the HTTP status code by throwing a [`GraphQLError` with an `http` extension](./errors/#setting-http-status-code-and-headers). For example:

Reviewer: I'd recommend we tuck this in its own section with a header:
```suggestion
#### Throwing errors in `context`

By default, if your `context` function throws an error, Apollo Server returns that error to the user in a JSON  response with an HTTP status code of 500. If the thrown error is not a `GraphQLError`, the error's message will be prepended with `"Context creation failed: "`.

You can control the HTTP status code of an error by throwing a [`GraphQLError` with an `http` extension](./errors/#setting-http-status-code-and-headers). For example:
```

}
```

#### Throwing errors in `context`

> By default, if your `context` function throws an error, Apollo Server returns that error to the user in a JSON  response with an HTTP status code of 500. If the thrown error is not a `GraphQLError`, the error's message will be prepended with `"Context creation failed: "`.

Reviewer 2: Extra whitespace

Author: After the colon in the string? That's intentional and accurate.
I see an extra space after JSON, I can fix that though I don't think it makes a difference when rendered...
